### PR TITLE
Issue #2622982: Change EntityManager to EntityTypeManager in GenericE…

### PR DESCRIPTION
**Pull request by DiDebru**
https://www.drupal.org/node/2622982

Hope it matches the current task

### CODE REVIEW

File: src/EventSubscriber/GenericEventSubscriber.php [hunk 1]
```diff
@@ -96,3 +96,4 @@ public function onRulesEvent(Event $event, $event_name) {
   }
 
 }
+a
```
Comment (klausi) [inline src/EventSubscriber/GenericEventSubscriber.php]: ?

File: src/Plugin/RulesAction/EntityCreateDeriver.php [hunk 2]
```diff
@@ -76,7 +76,7 @@ public function getDerivativeDefinitions($base_plugin_definition) {
       // other required base fields. This matches the storage create() behavior,
       // where only the bundle requirement is enforced.
       $bundle_key = $entity_type->getKey('bundle');
-      $base_field_definitions = $this->entityManager->getBaseFieldDefinitions($entity_type_id);
+      $base_field_definitions = $this->entityTypeManager->getBaseFieldDefinitions($entity_type_id);
```
Comment (klausi) [inline src/Plugin/RulesAction/EntityCreateDeriver.php]: but this should be EntityFieldManager, right?

File: src/Plugin/RulesAction/EntityCreateDeriver.php [hunk 3]
```diff
@@ -25,22 +26,28 @@ class EntityCreateDeriver extends DeriverBase implements ContainerDeriverInterfa
...
    */
-  protected $entityManager;
-
+  protected $entityTypeManager;
+  /**
+  * The entity field manager
+  *
+  * @var \Drupal\Core\Entity\EntityFieldManagerInterface;
+  */
+  protected $entityFieldManager
```
Comment (klausi) [inline src/Plugin/RulesAction/EntityCreateDeriver.php]: missing semi colon at the end

File: src/Plugin/RulesEvent/EntityInsertDeriver.php [hunk 4]
```diff
@@ -8,8 +8,9 @@
 namespace Drupal\rules\Plugin\RulesEvent;
 
 use Drupal\Component\Plugin\Derivative\DeriverBase;
+use Drupal\Core\Entity\EntityTypeManager;
```
Comment (klausi) [inline src/Plugin/RulesEvent/EntityInsertDeriver.php]: why is this use statement needed?

File: tests/src/Integration/RulesAction/RulesActionManagerTest.php [hunk 5]
```diff
@@ -9,6 +9,7 @@
 
 use Drupal\Tests\rules\Integration\RulesIntegrationTestBase;
 
+
```
Comment (klausi) [inline tests/src/Integration/RulesAction/RulesActionManagerTest.php]: unrelated change?

File: tests/src/Integration/RulesEntityIntegrationTestBase.php [hunk 6]
```diff
@@ -80,7 +80,6 @@ public function setUp() {
     ];
 
     $type_array = [];
-
```
Comment (klausi) [inline tests/src/Integration/RulesEntityIntegrationTestBase.php:83]: unrelated change?

File: src/Plugin/RulesEvent/EntityInsertDeriver.php (hunk 4 above)
Comment (DiDebru) [inline src/Plugin/RulesEvent/EntityInsertDeriver.php]: Just tried to implement it cause the tests failed i remove it and test again

File: tests/src/Integration/RulesEntityIntegrationTestBase.php (hunk 6 above)
Comment (DiDebru) [inline tests/src/Integration/RulesEntityIntegrationTestBase.php:83]: Thats from the merge conflict change i'll fix that.

**Comment by klausi** [comment]
merged, thanks!
